apps/src/templates/sectionProgressV2/SectionProgressV2.jsx

```diff
@@ -28,10 +30,26 @@ function SectionProgressV2({
     }
   }, [unitData, isLoadingProgress, isRefreshingProgress, scriptId, sectionId]);
 
+  React.useEffect(() => {
```

Comment:
Note: At one point I wanted to move all this logic to the `IconKey` component but given that would require passing all the `unitData` to the child component, putting the logic here felt like a better choice.